hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -88,6 +97,55 @@ findFaucetUTxO RunningNode{networkId, nodeSocket} lovelace = do
       FaucetHasNotEnoughFunds{faucetUTxO}
   pure foundUTxO
 
+seedFromFaucetBlockfrost ::
+  -- | Recipient of the funds
+  VerificationKey PaymentKey ->
+  -- | Amount to get from faucet
+  Coin ->
+  Blockfrost.BlockfrostClientT IO UTxO
+seedFromFaucetBlockfrost receivingVerificationKey lovelace = do
+  (faucetVk, faucetSk) <- liftIO $ keysFor Faucet
+
+  Blockfrost.Genesis
+    { Blockfrost._genesisNetworkMagic = networkMagic
+    , Blockfrost._genesisSystemStart = systemStart'
+    } <-
+    Blockfrost.queryGenesisParameters
+  pparams <- Blockfrost.queryProtocolParameters
+  let networkId = Blockfrost.toCardanoNetworkId networkMagic
+  let changeAddress = buildAddress faucetVk networkId
+  let receivingAddress = buildAddress receivingVerificationKey networkId
+  let theOutput =
+        TxOut
+          (shelleyAddressInEra shelleyBasedEra receivingAddress)
+          (lovelaceToValue lovelace)
+          TxOutDatumNone
+          ReferenceScriptNone
+  stakePools' <- Blockfrost.listPools
+  let stakePools = Set.fromList (Blockfrost.toCardanoPoolId <$> stakePools')
+  let systemStart = SystemStart $ posixSecondsToUTCTime systemStart'
+  eraHistory <- Blockfrost.queryEraHistory
+  foundUTxO <- findUTxO networkId changeAddress lovelace
+  case buildTransactionWithPParams' pparams systemStart eraHistory stakePools (mkVkAddress networkId faucetVk) foundUTxO [] [theOutput] of
+    Left e -> liftIO $ throwIO $ FaucetFailedToBuildTx{reason = e}
+    Right tx -> do
+      let signedTx = signTx faucetSk tx
+      eResult <- Blockfrost.tryError $ Blockfrost.submitTransaction signedTx
+      case eResult of
+        Left err -> liftIO $ throwIO $ FaucetBlockfrostError{blockFrostError = show err}
+        Right _ -> do
+          void $ Blockfrost.awaitUTxO networkId [changeAddress] (txId signedTx) 100
+          Blockfrost.awaitUTxO networkId [receivingAddress] (txId signedTx) 100
```

Comment:
if awaitUTxO takes a list of addresses, why not awaiting for both at once and then return the appropriate?

Reply:
true, I should do that.

Reply:
ah no. If I do that then the produced UTxO will have entries for the faucet key as well. In the test we need to get only the relevant pub key output UTxO.